config/core/300-resources/configuration.yaml

```diff
@@ -247,6 +247,19 @@ spec:
                                     description: Minimum consecutive failures for the probe to be considered failed after having succeeded. Defaults to 3. Minimum value is 1.
                                     type: integer
                                     format: int32
+                                  grpc:
+                                    description: GRPC specifies an action involving a GRPC port. This is a beta field and requires enabling GRPCContainerProbe feature gate.
+                                    type: object
+                                    required:
+                                      - port
+                                    properties:
+                                      port:
+                                        description: Port number of the gRPC service. Number must be in the range 1 to 65535.
+                                        type: integer
+                                        format: int32
+                                      service:
```

Comment:
The service field of grpc probe is optional but I have to enter an empty string(`""`) as shown below when apply gRPC probe to knative service. 

```yaml
        readinessProbe:
          grpc:
            port: 8080
            service: ""
```

Could you give me your opinion on this? 

The `service` field can be omitted when using Kubernetes core resource(e.g. deployment) instead of knative service.

Reply:
Failed test `istio-latest-no-mesh_serving_main` is caused by above reason..

```
=== FAIL: test/conformance/runtime TestProbeRuntime/grpc-period=1 (0.19s)
    readiness_probe_test.go:128: Creating a new Service
    service.go:150: Creating a new Service service probe-runtime-grpc-period-1-pkqhrzwo
    crd.go:36:  resource {<nil> <nil> <*>{&TypeMeta{Kind:,APIVersion:,} &ObjectMeta{Name:probe-runtime-grpc-period-1-pkqhrzwo,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[]OwnerReference{},Finalizers:[],ManagedFields:[]ManagedFieldsEntry{},} {{&ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[]OwnerReference{},Finalizers:[],ManagedFields:[]ManagedFieldsEntry{},}} {<nil>}} {{0 <nil> <nil>} { } {<nil> <nil> <nil>}}} <nil>}
    readiness_probe_test.go:137: Failed to create initial Service: probe-runtime-grpc-period-1-pkqhrzwo: Service.serving.knative.dev "probe-runtime-grpc-period-1-pkqhrzwo" is invalid: spec.template.spec.containers[0].readinessProbe.grpc.service: Invalid value: "null": spec.template.spec.containers[0].readinessProbe.grpc.service in body must be of type string: "null"
=== FAIL: test/conformance/runtime TestProbeRuntime/grpc (0.11s)
    readiness_probe_test.go:128: Creating a new Service
    service.go:150: Creating a new Service service probe-runtime-grpc-vasqolmt
    crd.go:36:  resource {<nil> <nil> <*>{&TypeMeta{Kind:,APIVersion:,} &ObjectMeta{Name:probe-runtime-grpc-vasqolmt,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[]OwnerReference{},Finalizers:[],ManagedFields:[]ManagedFieldsEntry{},} {{&ObjectMeta{Name:,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{},OwnerReferences:[]OwnerReference{},Finalizers:[],ManagedFields:[]ManagedFieldsEntry{},}} {<nil>}} {{0 <nil> <nil>} { } {<nil> <nil> <nil>}}} <nil>}
    readiness_probe_test.go:137: Failed to create initial Service: probe-runtime-grpc-vasqolmt: Service.serving.knative.dev "probe-runtime-grpc-vasqolmt" is invalid: spec.template.spec.containers[0].readinessProbe.grpc.service: Invalid value: "null": spec.template.spec.containers[0].readinessProbe.grpc.service in body must be of type string: "null"
```

Reply:
It seems like the defaulting is not showing up in our CRD spec here

https://pkg.go.dev/k8s.io/api@v0.27.3/core/v1#GRPCAction

A potential work around is to do that default in the revision go type we have a `SetDefaults`